firebase-mirror/src/main/java/io/spine/server/firebase/EventSubscriptionPublisher.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.firebase;
+
+import com.google.cloud.firestore.CollectionReference;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Message;
+import com.google.protobuf.util.Timestamps;
+import io.spine.base.Identifier;
+import io.spine.core.Event;
+import io.spine.core.EventContext;
+import io.spine.server.storage.StorageField;
+
+import java.util.Map;
+
+import static com.google.cloud.firestore.Blob.fromBytes;
+import static io.spine.protobuf.AnyPacker.unpack;
+import static io.spine.server.firebase.EntitySubscriptionPublisher.EntityStateField.bytes;
+import static io.spine.server.firebase.EntitySubscriptionPublisher.EntityStateField.id;
+import static io.spine.server.firebase.EventSubscriptionPublisher.EventStateField.producer_id;
+import static io.spine.server.firebase.EventSubscriptionPublisher.EventStateField.timestamp;
+
+/**
+ * A subscription update publisher for the event subscriptions.
+ *
+ * <p>The document identifier is an event ID and the composed document body contains the serialized
+ * event message as well as some meta-information.
+ *
+ * <p>See {@link EventStateField} for the details on storage format.
+ */
+final class EventSubscriptionPublisher extends FirestoreSubscriptionPublisher<Event> {
```

Comment:
If `FirestoreSubscriptionPublisher` is just `FirestorePublisher` (see my comment below), this is `EventPublisher`.